lib/identity/client.js

```diff
@@ -155,6 +155,27 @@ class Client extends PartialClient {
     return resultQuery
   }
   /**
+   * getLatestSecrets
+   *
+   * @param {string} secretName   The name of the Secret given by User.
+   * @param {string} secretType   The type of the Secret chosen by User.
+   *
+   *
+   * @return {Promise<Record>}
+   */
+  async getLatestSecret(secretName, secretType) {
+    const request = new Search(true)
+    let searchType = `tozny.secret.${SECRET_UUID}.${secretType}.${secretName}`
+    request.match({ type: searchType })
+    let resultQuery = await this.storage.search(request)
+    let resultList = await resultQuery.next()
+    let lengthOfResults = resultList.length
+    if (resultList.length != 0) {
+      return { exists: true, result: resultList[lengthOfResults - 1] }
+    }
+    return { exists: false, result: resultList[0] }
```

Comment:
can result be null instead / won't this blow up if the list is empty and you try and index the first item (which won't exist because its empty...)